two_factor/admin.py

```diff
@@ -30,9 +33,25 @@ def has_permission(self, request):
     def login(self, request, extra_context=None):
         """
         Redirects to the site login page for the given HttpRequest.
+        If user has admin permissions but 2FA not setup, then redirect to
+            2FA setup page.
         """
+        # redirect to admin page after login
         redirect_to = request.POST.get(REDIRECT_FIELD_NAME, request.GET.get(REDIRECT_FIELD_NAME))
 
+        # if user (is_active and is_staff)
+        if request.method == "GET" and AdminSite().has_permission(request):
```

Comment:
Why not `self.has_permission(request)`?

Reply:
`self.has_permission()` func checks if user (is_staff, is_active and is_verified) . 
`AdminSite().has_permission()`  checks if user (is_staff, is_active ). Then based on if user is not verified (OTP not setup) we want to redirect to OTP setup page.

Reply:
`super()` should be used instead then. This is a mixin and we don't know what other checks users might have/want so there might be other classes involved, not just `AdminSite`.